examples/model_selection/plot_roc.py

```diff
@@ -135,55 +272,177 @@
+        ax=ax,
+        name=f"ROC curve for {label_a} as positive class",
+    )
+    RocCurveDisplay.from_predictions(
+        b_true,
+        y_score[ab_mask, idx_b],
+        ax=ax,
+        name=f"ROC curve for {label_b} as positive class",
+    )
+    plt.plot([0, 1], [0, 1], "k--", label="ROC curve for chance level (AUC = 0.5)")
+    plt.xlim([0.0, 1.0])
+    plt.ylim([0.0, 1.0])
+    plt.axis("square")
+    plt.xlabel("False Positive Rate")
+    plt.ylabel("True Positive Rate")
+    plt.title(f"{target_names[idx_a]} vs {label_b} ROC curves")
+    plt.legend()
+    plt.show()
+
+print(f"Macro-averaged One-vs-One ROC AUC score:\n{np.average(pair_scores):.2f}")
+
+# %%
+# One can also assert that the macro-average we computed "by hand" is equivalent
+# to the implemented `average="macro"` option of the
+# :class:`~sklearn.metrics.roc_auc_score` function.
+
+macro_roc_auc_ovo = roc_auc_score(
+    y_onehot_test,
+    y_score,
+    multi_class="ovo",
+    average="macro",
 )
-print(
-    "One-vs-Rest ROC AUC scores:\n{:.6f} (macro),\n{:.6f} "
-    "(weighted by prevalence)".format(macro_roc_auc_ovr, weighted_roc_auc_ovr)
+
+print(f"One-vs-One ROC AUC scores:\n{macro_roc_auc_ovo:.2f}")
+
+# %%
+# Plot all OvO ROC curves together
+# --------------------------------
+
+ovo_tpr = np.zeros_like(fpr_grid)
+for ix, (label_a, label_b) in enumerate(pair_list):
+    ovo_tpr += mean_tpr[ix]
+    plt.plot(
+        fpr_grid,
+        mean_tpr[ix],
+        label=f"Mean {label_a} vs {label_b} ROC curve (AUC = {pair_scores[ix]:.2f})",
+    )
+
+ovo_tpr /= sum(1 for pair in enumerate(pair_list))
+
+plt.plot(
+    fpr_grid,
+    ovo_tpr,
+    label=f"One-vs-One macro-average ROC curve (AUC = {macro_roc_auc_ovo:.2f})",
+    linestyle=":",
+    linewidth=4,
 )
+plt.plot([0, 1], [0, 1], "k--", label="ROC curve for chance level (AUC = 0.5)")
+plt.xlim([0.0, 1.0])
+plt.ylim([0.0, 1.0])
+plt.axis("square")
+plt.xlabel("False Positive Rate")
+plt.ylabel("True Positive Rate")
+plt.legend()
+plt.show()
+
+# %%
+# We confirm that the classes "versicolor" and "virginica" are not well
+# identified by a linear classifier. Notice that the "virginica"-vs-the-rest
+# ROC-AUC score (0.77) is between the OvO ROC-AUC scores for "versicolor" vs
+# "virginica" (0.64) and "setosa" vs "virginica" (0.90). Indeed, the OvO
+# strategy gives additional information on the overlap between classes, at the
+# expense of computational cost.
+#
+# The OvO strategy is recommended if the user is mainly interested in correctly
+# identifying a particular class or subset of classes, whereas evaluating the
+# global performance of a classifier can be correctly resumed by a given
+# averaging strategy.
+#
+# Micro-average is dominated by the more frequent class, since the counts are
+# pooled. The macro-average better reflects the statistics of the less frequent
+# classes, and then is more appropriate when performance on all the classes is
+# equally important.
```

Comment:
```suggestion
# Micro-averaged OvR ROC is dominated by the more frequent class, since the
# counts are pooled. The macro-averaged alternative better reflects the
# statistics of the less frequent classes, and then is more appropriate when
# performance on all the classes is deemed equally important.
```